src/cls/refcount/cls_refcount_chunk.cc

```diff
@@ -0,0 +1,45 @@
+// -*- mode:C; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
```

Comment:
I think we should aim to have chunk_write_or_get, chunk_put, and chunk_read_refcount all in the same cls.  When I suggested 'cas' I expected them all to be there, but I forgot that the manifest can also refcount a non-cas object.  

The main thing that worries me is that the existing methods in cls_refcount are already used by rgw and we have to be careful not to break them.  On the other hand, the refcounts I expect for the cas objects are a bit different, where I expect we'll have some objects will millions of refs and we want to do some clever things with the way those are efficiently represented and scrubbed.  This makes me want a clean cls_cas... except for the non-fingerprint objects that aren't actually content-addressibly named.  Maybe cls_chunk captures the spirit of both?  And lets us pull *all* of the chunk refcounting into a single class without recursively calling into other cls_refcount methods.

I'm sorry to keep moving the goal posts around on this!  If you want we can sketch this out in a bit more detail first before adjusting the code again.  The main thing I want to ensure is that other consumers of the cas pool (besides the tiering code... probably rgw in the nearish future) have a simple and clean interface to consume.